operator/pkg/component/component.go

```diff
@@ -12,264 +12,220 @@
-	c.started = true
-	return nil
-}
-
-func (c *IstioComponent) RenderManifest() (string, error) {
-	return renderManifest(c)
-}
-
-// NewCoreComponent creates a new IstioComponent with the given componentName and options.
-func NewCoreComponent(cn name.ComponentName, opts *Options) *IstioComponent {
-	var component *IstioComponent
-	switch cn {
-	case name.IstioBaseComponentName:
-		component = NewCRDComponent(opts)
-	case name.PilotComponentName:
-		component = NewPilotComponent(opts)
-	case name.CNIComponentName:
-		component = NewCNIComponent(opts)
-	case name.IstiodRemoteComponentName:
-		component = NewIstiodRemoteComponent(opts)
-	case name.ZtunnelComponentName:
-		component = NewZtunnelComponent(opts)
-	default:
-		scope.Errorf("Unknown component componentName: " + string(cn))
-	}
-	return component
-}
-
-// NewCRDComponent creates a new IstioComponent and returns a pointer to it.
-func NewCRDComponent(opts *Options) *IstioComponent {
-	return &IstioComponent{
-		Options:       opts,
-		ComponentName: name.IstioBaseComponentName,
-	}
-}
-
-// NewPilotComponent creates a new IstioComponent and returns a pointer to it.
-func NewPilotComponent(opts *Options) *IstioComponent {
-	cn := name.PilotComponentName
-	return &IstioComponent{
-		Options:       opts,
-		ComponentName: cn,
-		ResourceName:  opts.Translator.ComponentMaps[cn].ResourceName,
-	}
-}
-
-// NewCNIComponent creates a new IstioComponent and returns a pointer to it.
-func NewCNIComponent(opts *Options) *IstioComponent {
-	cn := name.CNIComponentName
-	return &IstioComponent{
-		Options:       opts,
-		ComponentName: cn,
-	}
-}
-
-// NewIstiodRemoteComponent creates a new IstioComponent and returns a pointer to it.
-func NewIstiodRemoteComponent(opts *Options) *IstioComponent {
-	cn := name.IstiodRemoteComponentName
-	return &IstioComponent{
-		Options:       opts,
-		ComponentName: cn,
-	}
-}
-
-// NewIngressComponent creates a new IstioComponent and returns a pointer to it.
-func NewIngressComponent(resourceName string, index int, spec *v1alpha1.GatewaySpec, opts *Options) *IstioComponent {
-	return &IstioComponent{
-		Options:       opts,
-		ComponentName: name.IngressComponentName,
-		ResourceName:  resourceName,
-		index:         index,
-		componentSpec: spec,
+	// ContainerName maps a Name to the name of the container in a Deployment.
+	ContainerName string
+	// HelmSubdir is a mapping between a component name and the subdirectory of the component Chart.
+	HelmSubdir string
+	// ToHelmValuesTreeRoot is the tree root in values YAML files for the component.
+	ToHelmValuesTreeRoot string
+	// FlattenValues, if true, means the component expects values not prefixed with ToHelmValuesTreeRoot
+	// For example `.name=foo` instead of `.component.name=foo`.
+	FlattenValues bool
+	// AltEnablementPath is the alternative path, from values, to enable the component
+	AltEnablementPath string
+}
+
+func (c Component) Get(merged values.Map) ([]apis.GatewayComponentSpec, error) {
```

Comment:
It's a little confusing that it returns a GatewayComponentSpec, can we add a comment saying that this maybe isn't actually a gateway component?